airflow/www/views.py

```diff
@@ -258,7 +258,11 @@ def task_group_to_grid(task_item_or_group, dag, dag_runs, session):
     if isinstance(task_item_or_group, AbstractOperator):
         return {
             'id': task_item_or_group.task_id,
-            'instances': [wwwutils.get_task_summary(dr, task_item_or_group, session) for dr in dag_runs],
+            'instances': [
+                ts
+                for ts in (wwwutils.get_task_summary(dr, task_item_or_group, session) for dr in dag_runs)
+                if ts is not None
+            ],
```

Comment:
just to throw out another option,

```suggestion
            'instances': filter(
                lambda x: x is not None, 
                (wwwutils.get_task_summary(dr, task_item_or_group, session) for dr in dag_runs),
            ),
```

don't know if it also needs to be wrapped with `list`.  i know we tend to prefer comprehension over map etc, but in this case it saves you that `ts` variable, which, since we're already iterating over DRs, is a tiny bit more cognitive load i thought.

Reply:
FWIW if we don’t need to care about empty summary (or if we’re OK to also filter those out), we can simply do

```python
filter(None, (...))
```

Reply:
nice, i was thinking there must be a shorthand of some kind for this

Reply:
Alternative fix here: https://github.com/apache/airflow/pull/23932 - I think we shoudl slowly start add typing also to views.py as this precisely the kind of errors that would have been prevented if we had typing here in the first place.

Reply:
More type checking is always nice. I've been meaning to add typescript to the UI.

Reply:
> More type checking is always nice. I've been meaning to add typescript to the UI.

Oh yeah. Absolutely. Typescript is soooo much better than plain javascript. One thing that it makes easier - is for anyone to contribute more fixes in an easy way. (once we have a built-in automation to transpile it, which is super-easy with yarn etc). I think lack of typing support, autocomplete and verification makes it really difficult for "newcomers" to reason about their changes and it's so much easier to contribute small fixes there. 

You somehow loose that when you get familiar, when you put yourselve in the shoes of new contributor - it's a world of difference.